CHANGES.rst

```diff
@@ -120,6 +120,11 @@ astropy.nddata
 
 - Add a way for technically invalid but unambiguous units in a fits header to be
   parsed by ``CCDData``. [#9397]
+- ``NDData`` now only accepts WCS objects which implement either the high, or
+  low level APE 14 WCS API. All WCS objects are converted to a high level WCS
+  object, so ``NDData.wcs`` now always returns a high level APE 14 object. Not
+  all array slices are valid for wcs objects, so some slicing operations which
+  used to work may now fail. [#9067]
```

Comment:
I guess the proposal to simply "deprecate" the current behavior and removing it later instead of having a hard break now is out of the question?

Reply:
I'm inclined to go ahead with the hard break.

Reply:
Agree